src/Fixer/Casing/NativeFunctionTypeHintCasingFixer.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Casing;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\TypeAnalysis;
+use PhpCsFixer\Tokenizer\Analyzer\FunctionsAnalyzer;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ */
+final class NativeFunctionTypeHintCasingFixer extends AbstractFixer
+{
+    /**
+     * https://secure.php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration.
+     *
+     * self     PHP 5.0.0
+     * array    PHP 5.1.0
+     * callable PHP 5.4.0
+     * bool     PHP 7.0.0
+     * float    PHP 7.0.0
+     * int      PHP 7.0.0
+     * string   PHP 7.0.0
+     * iterable PHP 7.1.0
+     * void     PHP 7.1.0
+     * object   PHP 7.2.0
+     *
+     * @var array<string, true>
+     */
+    private $hints;
+
+    /**
+     * @var FunctionsAnalyzer
+     */
+    private $functionsAnalyzer;
+
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->hints = [
+            'array' => true,
+            'callable' => true,
+            'self' => true,
+        ];
+
+        if (\PHP_VERSION_ID >= 70000) {
+            $this->hints = array_merge(
+                $this->hints,
+                [
+                    'bool' => true,
+                    'float' => true,
+                    'int' => true,
+                    'string' => true,
+                ]
+            );
+        }
+
+        if (\PHP_VERSION_ID >= 70100) {
+            $this->hints = array_merge(
+                $this->hints,
+                [
+                    'iterable' => true,
+                    'void' => true,
+                ]
+            );
+        }
+
+        if (\PHP_VERSION_ID >= 70200) {
+            $this->hints = array_merge($this->hints, ['object' => true]);
+        }
+
+        $this->functionsAnalyzer = new FunctionsAnalyzer();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Type hints (of functions) defined by PHP should be called using the correct casing.',
```

Comment:
I think this would read better:

```suggestion
            'Native type hints for functions should use the correct case.',
```